### PR TITLE
docs(localize): add warning about ICU not working with $localize, adap…

### PR DESCRIPTION
**Pull request by 000panther**
…t example

Since I fell over it i added a warning that ICU will not work in $localize tags (this was not obvious for me from the documentation) and adapted the first example to show that it is used from a template.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Documentation not mentioning ICU does not work with $localize

## What is the new behavior?
Documention mentions that ICU does only work in templates.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by JeanMeche** [comment]
Hi, for you commit to be valid, you need to follow [the guidelines](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit). 
In your case, you're missing the colon after the parenthesis !

File: aio/content/guide/i18n-common-prepare.md [hunk 1]
```diff
@@ -229,11 +229,20 @@ That one translation entry is merged back into the application wherever the same
...
+  
+<!--todo: replace with code-example -->
 
-{ component_property, icu_clause, case_statements }
+In the example we assume that the variable numberOfItems exists in the template and use the plural function to render correct pluralization.
+  
+<code-example>  
+  
+`<span i18n>
+  Item count: {numberOfItems, plural, =0 { No item } =1 { One item } other { {{numberOfItems }} items }} }
```
Comment (JeanMeche) [inline aio/content/guide/i18n-common-prepare.md:244]:
```suggestion
  Item count: {numberOfItems, plural, =0 { No item } =1 { One item } other { {{ numberOfItems }} items }} }
```

**Review by JeanMeche — COMMENTED** [review]
2 nits from me ! 

Thank you for improving the docs btw !

File: aio/content/guide/i18n-common-prepare.md [hunk 2]
```diff
@@ -229,11 +229,20 @@ That one translation entry is merged back into the application wherever the same
 ICU expressions help you mark alternate text in component templates to meet conditions.
 An ICU expression includes a component property, an ICU clause, and the case statements surrounded by open curly brace \(`{`\) and close curly brace \(`}`\) characters.
 
-<!--todo: replace with code-example -->
+<div class="alert is-helpful">
+ICU expressions will only work in templates, not when using $localize in other angular code.
```
Comment (AndrewKushnir) [inline aio/content/guide/i18n-common-prepare.md:233]:
```suggestion
ICU expressions will only work in Angular templates. ICUs are **not** supported in $localize calls in TypeScript code.   
```

**Review by AndrewKushnir — COMMENTED** [review]
@000panther thanks for improving Angular docs!

**Comment by jessicajaniuk** [comment]
@000panther Can you squash you commits down to one with the right commit header format? That should resolve the lint issue. If you rebase to main, as well, you'll likely resolve the size check, too.

**Comment by JeanMeche** [comment]
Kind ping, can you squash your commits  and follow the commit message guidelines ?

**Comment by jessicajaniuk** [comment]
Closing this since there's been no updates. Feel free to open a new PR if you'd like to see these changes land.

**Pull request closed without merging**